htdocs/core/class/html.formmail.class.php

```diff
@@ -330,7 +330,7 @@ public function get_form($addfileaction = 'addfile', $removefileaction = 'remove
 		$langs->loadLangs(array('other', 'mails'));
 
 		// Clear temp files. Must be done at beginning, before call of triggers
-		if (GETPOST('mode', 'alpha') == 'init' || (GETPOST('modelmailselected', 'alpha') && GETPOST('modelmailselected', 'alpha') != '-1'))
+		if (GETPOST('mode', 'alpha') == 'init' && (GETPOST('modelmailselected', 'alpha') && GETPOST('modelmailselected', 'alpha') != '-1'))
```

Comment:
Don't understand why a &. Doing that means the list of files will no more be clean whe we call the page with param mode = 'init'

Reply:
I was unsure of that, but, on "init" shouldn't the "modelmailselected" unset or set to 0 or -1 ?

Reply:
Actually the list is cleared each time the mailmodelselected parameter is not set to "-1" wich is very strange to me.